accord-core/src/main/java/accord/primitives/KeyRanges.java

```diff
@@ -27,22 +28,24 @@
 import java.util.*;
 
 import static accord.utils.SortedArrays.Search.FAST;
+import static accord.utils.Utils.toArray;
 
 public class KeyRanges implements Iterable<KeyRange>
 {
     public static final KeyRanges EMPTY = ofSortedAndDeoverlappedUnchecked(new KeyRange[0]);
 
     final KeyRange[] ranges;
 
-    private KeyRanges(KeyRange[] ranges)
+    @VisibleForImplementation
```

Comment:
Why is this being made public? Can we use the static constructor methods that enforce the constraints?